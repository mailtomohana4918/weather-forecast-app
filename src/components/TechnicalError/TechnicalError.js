import { Typography, Box, Card, CardMedia, CardContent } from '@material-ui/core'
import { makeStyles } from "@material-ui/core/styles";
import React from "react"
import weather from '../../weather.jpg'

const useStyles = makeStyles(() => ({
    boxContainer: {
        paddingTop: "24px",
        width: "600px",
        paddingLeft: "18px",
        paddingRight: "18px"
    }
}));

//component renders when there is a error response from the server
export const TechnicalError = () => {
    const classes = useStyles();
    return (
        <Box className={classes.boxContainer}>
            <Card sx={{ maxWidth: 500 }}>
                <CardMedia
                    component="img"
                    height="140"
                    image={weather}
                    alt="weather"
                />
                <CardContent>
                    <Typography gutterBottom variant="h5">
                        Sorry, there's been an error
                    </Typography>

                    <Typography variant='body2'>We can't do what you've asked right now due to an error on our end.</Typography>
                    <Typography variant='body2'>Please try again later.</Typography>
                </CardContent>
            </Card>
        </Box>
    )
}



