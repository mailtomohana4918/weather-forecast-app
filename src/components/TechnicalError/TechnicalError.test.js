import { render } from "@testing-library/react";
import React from "react";
import {TechnicalError} from "./TechnicalError";

describe("TechnicalError Component", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<TechnicalError />);
    expect(asFragment()).toMatchSnapshot();
  });
});