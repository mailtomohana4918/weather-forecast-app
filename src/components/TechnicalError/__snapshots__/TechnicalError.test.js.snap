// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnicalError Component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-boxContainer-1"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      sx="[object Object]"
    >
      <img
        alt="weather"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        height="140"
        src="weather.jpg"
      />
      <div
        class="MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Sorry, there's been an error
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          We can't do what you've asked right now due to an error on our end.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Please try again later.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
